### PR TITLE
Add Huawei OBS integration.

### PR DESCRIPTION
**Pull request by uronce-cc**

### CODE REVIEW

File: underfs/obs/pom.xml [hunk 1]
```diff
@@ -0,0 +1,216 @@
+<!--
+  ~ The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
```
Comment (madanadit) [inline underfs/obs/pom.xml]: is that ~ a typo

File: underfs/obs/src/main/java/alluxio/underfs/oss/OBSOutputStream.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    Preconditions.checkArgument(bucketName != null && !bucketName.isEmpty(),
+        "Bucket name must not be null or empty.");
+    Preconditions.checkArgument(key != null && !key.isEmpty(),
+        "OBS object key must not be null or empty.");
+    Preconditions.checkArgument(client != null, "ObsClient must not be null.");
+    mBucketName = bucketName;
+    mKey = key;
+    mObsClient = client;
+
+    mFile = new File(PathUtils.concatPath("/tmp", UUID.randomUUID()));
```
Comment (madanadit) [inline underfs/obs/src/main/java/alluxio/underfs/oss/OBSOutputStream.java]: do we have a requirement that '/tmp' is accessible by Alluxio servers?. if yes, please call it out in the docs

File: underfs/obs/src/main/java/alluxio/underfs/oss/OBSOutputStream.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    mLocalOutputStream.flush();
+  }
+
+  /**
+   * Closes this output stream. When an output stream is closed, the local temporary file is
+   * uploaded to OSS Service. Once the file is uploaded, the temporary file is deleted.
+   */
+  @Override
+  public void close() throws IOException {
+    if (mClosed.getAndSet(true)) {
```
Comment (madanadit) [inline underfs/obs/src/main/java/alluxio/underfs/oss/OBSOutputStream.java]: does it make sense to log warning if closing an already closed stream?

File: underfs/obs/pom.xml (hunk 1 above)
Comment (uronce-cc) [reply underfs/obs/pom.xml]: done

File: underfs/obs/src/main/java/alluxio/underfs/oss/OBSOutputStream.java (hunk 2 above)
Comment (uronce-cc) [reply underfs/obs/src/main/java/alluxio/underfs/oss/OBSOutputStream.java]: replaced this line with File.createTempFile

File: underfs/obs/src/main/java/alluxio/underfs/oss/OBSOutputStream.java (hunk 3 above)
Comment (uronce-cc) [reply underfs/obs/src/main/java/alluxio/underfs/oss/OBSOutputStream.java]: added

**Comment by uronce-cc** [comment]
@madanadit Thanks for the comments, PTAL.

**Pull request merged**